scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua

```diff
@@ -20,25 +20,25 @@ require("scripts/zones/Windurst_Waters/TextIDs");
 -----------------------------------
 
 function onTrade(player,npc,trade)
-	FoodForThought = player:getQuestStatus(WINDURST,FOOD_FOR_THOUGHT);
-	KenapaFood = player:getVar("Kenapa_Food_var"); -- Variable to track progress of Kenapa-Keppa in Food for Thought
-	
-	if (FoodForThought == QUEST_ACCEPTED) then
-		count = trade:getItemCount();
-		gil = trade:getGil();
-		if (trade:hasItemQty(4409,1) == false) then -- Traded in wrong item. Not accepted.
-			player:startEvent(0x0149);
-		elseif (count == 1 and trade:hasItemQty(4409,1) == true and gil == 0 and KenapaFood == 0) then -- Traded item without receiving order
-			rand = math.random(1,2);
-			if (rand == 1) then
-				player:startEvent(0x014b);
-			else
-				player:startEvent(0x014a,120);
-			end
-		elseif (count == 1 and trade:hasItemQty(4409,1) == true and gil == 0 and KenapaFood == 3) then	-- Traded item after receiving order
-			player:startEvent(0x0147,120);
-		end
-	end
+    FoodForThought = player:getQuestStatus(WINDURST,FOOD_FOR_THOUGHT);
+    KenapaFood = player:getVar("Kenapa_Food_var"); -- Variable to track progress of Kenapa-Keppa in Food for Thought

```

Comment:
i know this wasn't part of your change, but make FoodForThought and KenapaFood local variables (local as in lua local, not dsp localvar)